internal/pkg/infrastructure/repositories/database/database.go

```diff
@@ -103,6 +104,29 @@ func NewDatabaseConnection(connect ConnectorFunc) (Datastore, error) {
 	return db, nil
 }
 
+func (db *myDB) GetWaterConsumptions(deviceId string, from, to time.Time, limit uint64) ([]models.WaterConsumption, error) {
+	wcos := []models.WaterConsumption{}
+	gorm := db.impl.Order("timestamp")
+
+	if deviceId != "" {
+		gorm = gorm.Where("device = ?", deviceId)
+	}
+
+	if !from.IsZero() || !to.IsZero() {
+		gorm = insertTemporalSQL(gorm, "timestamp", from, to) //make sure to set to Descending order
```

Comment:
Make sure to set descending order where and when? I think we need a test that shows the behaviour of this function when from and to are zero, but there is a limit. That might be when you need descending order in the query.